craft_application/application.py

```diff
@@ -491,6 +494,19 @@ def _pre_run(self, dispatcher: craft_cli.Dispatcher) -> None:
                     resolution="Ensure the path entered is correct.",
                 )
 
+    def get_arg_or_config(
+        self, parsed_args: argparse.Namespace, item: str
+    ) -> Any:  # noqa: ANN401
+        """Get a configuration option that could be overridden by a command argument.
+
+        :param parsed_args: The argparse Namespace to check.
+        :param item: the name of the namespace or config item.
+        :returns: the requested value.
+        """
+        if hasattr(parsed_args, item):
+            return getattr(parsed_args, item)
+        return self.services.config.get(item)
```

Comment:
Theoretically, you could use the overloaded version of `getattr` here, but I don't have a strong preference
```suggestion
        return getattr(parsed_args, item, self.services.config.get(item))
```